source/solvers/navier_stokes_vof_assemblers.cc

```diff
@@ -584,6 +584,76 @@ template class GLSNavierStokesVOFAssemblerSTF<2>;
 template class GLSNavierStokesVOFAssemblerSTF<3>;
 
 
+template <int dim>
+void
+GLSNavierStokesVOFAssemblerMarangoni<dim>::assemble_matrix(
+  NavierStokesScratchData<dim> & /*scratch_data*/,
+  StabilizedMethodsTensorCopyData<dim> & /*copy_data*/)
+{}
+
+template <int dim>
+void
+GLSNavierStokesVOFAssemblerMarangoni<dim>::assemble_rhs(
+  NavierStokesScratchData<dim> &        scratch_data,
+  StabilizedMethodsTensorCopyData<dim> &copy_data)
+{
+  // Surface tension gradient
+  const double surface_tension_gradient =
+    STF_properties.surface_tension_gradient;
+
+  // Loop and quadrature informations
+  const auto &       JxW        = scratch_data.JxW;
+  const unsigned int n_q_points = scratch_data.n_q_points;
+  const unsigned int n_dofs     = scratch_data.n_dofs;
+
+  // Copy data elements
+  auto &strong_residual = copy_data.strong_residual;
+  auto &local_rhs       = copy_data.local_rhs;
+
+  // Loop over the quadrature points
+  for (unsigned int q = 0; q < n_q_points; ++q)
+    {
+      // Gather filtered phase fraction gradient
+      const Tensor<1, dim> &filtered_phase_fraction_gradient_value =
+        scratch_data.filtered_phase_fraction_gradient_values[q];
+
+      const double phase_fraction_gradient_norm =
+        filtered_phase_fraction_gradient_value.norm();
+
+      const Tensor<1, dim> normalized_filtered_phase_fraction_gradient =
+        filtered_phase_fraction_gradient_value /
+        (phase_fraction_gradient_norm + DBL_MIN);
+
+      // Gather temperature gradient
+      const Tensor<1, dim> temperature_gradient =
+        scratch_data.temperature_gradients[q];
+
+      const double JxW_value = JxW[q];
+
+      const Tensor<1, dim> tmp_marangoni =
+        -1.0 * surface_tension_gradient *
+        (temperature_gradient - normalized_filtered_phase_fraction_gradient *
+                                  (normalized_filtered_phase_fraction_gradient *
+                                   temperature_gradient)) *
+        phase_fraction_gradient_norm;
```

Comment:
This equation does not agree with the one written in the header file because of the last  | \nabla \phi | term . Is it like this (header file)?:

```
 * $$\mathbf{F_{Ma}}= \frac{\partial \sigma}{\partial T} \left[ \nabla T
 * - \frac{\nabla \phi}{| \nabla \phi |} \left( \frac{ \nabla \phi }
 * {| \nabla \phi |} \cdot \nabla T \right) | \nabla \phi |  \right]
```
 
Or like this (this source file)?: 
```
 * $$\mathbf{F_{Ma}}= \frac{\partial \sigma}{\partial T} \left[ \nabla T
 * - \frac{\nabla \phi}{| \nabla \phi |} \left( \frac{ \nabla \phi }
 * {| \nabla \phi |} \cdot \nabla T \right)  \right]  | \nabla \phi |
```

Reply:
Very good comment. The source file is correct, I updated the header file. The density ratio term was also missing and it is introduced.